docs/cookbook/recipe_creating_a_locale_switcher.rst

```diff
@@ -0,0 +1,161 @@
+Creating a locale switcher
+==========================
+
+This is a full working example of creating a locale switcher.
+
+.. note::
+    This article assumes you are using Symfony 4, Flex, and autowiring.
+
+Prerequisites
+-------------
+
+Install the following Sonata dependencies:
+
+* `Sonata Easy-Extends Bundle`_
+* `Sonata Intl Bundle`_
+
+Create a ``LocaleSubscriber`` and a ``UserLocaleSubscriber`` class as explained `here`_.
+
+.. note::
+    Sonata Intl Bundle is optional, but will allow list of locales to be shown as list of languages.
+
+Handle locale switch
+--------------------
+
+Create a controller to handle locale switch in ``src/Controller/LocaleController.php``:
+
+.. code-block:: php
+
+    <?php
+
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\Routing\Annotation\Route;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Session\SessionInterface;
+    use Symfony\Component\HttpFoundation\RedirectResponse;
+
+    final class LocaleController
+    {
+        /**
+        * @Route("/locale/{locale}", name="locale")
+        */
+        public function index(Request $request, SessionInterface $session, string $locale): RedirectResponse
+        {
+            $session->set('_locale', $locale);
+            $referer = $request->headers->get('referer');
+            return new RedirectResponse($referer ?: '/');
+        }
+    }
+
+Extend SonataAdminBundle
+------------------------
+
+Open a command console, enter your project directory and execute the
+following command:
+
+.. code-block:: bash
+
+    $ bin/console sonata:easy-extends:generate SonataAdminBundle --dest=src --namespace_prefix=App
+
+Register the bundle in the kernel of your application:
+
+.. code-block:: php
+
+    return [
+        ...
+        App\Application\Sonata\AdminBundle\ApplicationSonataAdminBundle::class => ['all' => true],
```

Comment:
Seems no, I will update.

> Why do you use docs for a feature? Can you provide an implementation instead or create a separate bundle for it.

Isn't it a very small functionnality for a bundle? About implementation I don't know, I am very new to Sonata and Symfony, but it could be very good exercise...

Reply:
If you drop easy admin part of the docs, you don't need this anymore

Reply:
True, already done in last commit.

Reply:
Any opinion about Bundle or implementation?